pkg/ccl/backupccl/backup_test.go

```diff
@@ -5591,6 +5591,84 @@ func TestBackupCreatedStatsFromIncrementalBackup(t *testing.T) {
 	sqlDB.CheckQueryResults(t, getStatsQuery(`"data 2".bank`), statsBackup2)
 }
 
+// TestProtectedTimestampSpanSelectionDuringBackup tests that the expected spans
+// to timestamp protect are selected during a backup. It does not rigorously
+// test the correctness of protecting the timestamp as this is covered by
+// TestProtectedTimestampsDuringBackup.
+func TestProtectedTimestampSpanSelectionDuringBackup(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	allowRequest := make(chan struct{})
+	dir, dirCleanupFn := testutils.TempDir(t)
+	defer dirCleanupFn()
+	params := base.TestClusterArgs{}
+	params.ServerArgs.ExternalIODir = dir
+	params.ServerArgs.Knobs.Store = &kvserver.StoreTestingKnobs{
+		TestingRequestFilter: func(ctx context.Context, ba roachpb.BatchRequest) *roachpb.Error {
+			for _, ru := range ba.Requests {
+				switch ru.GetInner().(type) {
+				case *roachpb.ExportRequest:
+					<-allowRequest
+				}
+			}
+			return nil
+		},
+	}
+	tc := testcluster.StartTestCluster(t, 3, params)
+	defer tc.Stopper().Stop(ctx)
+
+	tc.WaitForNodeLiveness(t)
+	require.NoError(t, tc.WaitForFullReplication())
+
+	close(allowRequest)
+	conn := tc.ServerConn(0)
+	runner := sqlutils.MakeSQLRunner(conn)
+	runner.Exec(t, "CREATE DATABASE test; USE test;")
+	runner.Exec(t, "CREATE TABLE foo (k INT PRIMARY KEY, v BYTES, INDEX bar (v))")
+
+	// Create an interleave hierarchy of tables.
```

Comment:
Do contiguous table spans get merged together? I feel like they should. Can you add a few more tables to the end here and see that they contribute exactly one more span?